message-format.md

```diff
@@ -1,47 +1,106 @@
+| `val`     | Any          | `3.33`                                    | Yes      | Payload of the message of which type is defined in `val_t` property.                                                                                                                                                                 |
+| `storage` | Object       | `{"sub_value": "kWh"}`                    | No       | Defines optional storage aggregation key for multi-value reports. Applies only to device reports. Parser must accept missing key or `null` value as an empty `object`.                                                               |
+| `props`   | String Map   | `{"unit":"W"}`                            | No       | An optional map of properties. Please note that all properties including numeric ones must be encoded as strings. Parser must accept missing key or `null` value as an empty `map`.                                                  |
+| `tags`    | String Array | `{"test"}`                                | No       | An optional list of tags. Parser must accept missing key or `null` value as an empty `array`.                                                                                                                                        |
+| `resp_to` | String       | `"pt:j1/mt:rsp/rt:app/rn:angry-dog/ad:1"` | No       | A topic where a response to the request will be expected. Does not apply to most of the broadcast type of events, especially asynchronous device events for which adapter will ignore this property. Parser must accept missing key. |
+| `corid`   | String       | `"e0749aaa-05b8-4983-89e1-f225b539cc40"`  | No       | Response correlation identifier that is equal to the `uid` property of the request, used for request-response matching. Parser must accept missing key.                                                                              |
+| `uid`     | String       | `"9bb1be75-35d7-4069-ac00-b974315f7ec3"`  | Yes      | A unique message identifier in form of UUID. It is recommended to use Version 4.                                                                                                                                                     |
+| `ctime`   | String       | `"2022-12-02T10:08:27.5+01:00"`           | Yes      | Message creation time in RFC3339 format. For compatibility reasons a message parser must treat a missing key as the current timestamp. See [time format](#time-format) section for disambiguation.                                   |
+| `src`     | String       | `"smarthome-app"`                         | Yes      | Name of the message publisher identifying a HUB, mobile or cloud application or service. For compatibility reasons a message parser must tolerate a missing key.                                                                     |
```

Comment:
mobile, cloud, application or a service.

Reply:
Not really, but reworded to make it more intelligible.